semgrep/semgrep/evaluation.py

```diff
@@ -486,13 +459,11 @@ def _evaluate_expression(
     steps_for_debugging: List[Dict[str, Any]],
     flags: Optional[Dict[str, Any]] = None,
 ) -> Set[Range]:
-    if (
-        expression.operator == OPERATORS.AND_EITHER
-        or expression.operator == OPERATORS.AND_ALL
-    ):
-        assert (
```

Comment:
Same as above, if we care about this `assert` we should perform a real check and `raise` if necessary.